org.adempiere.base/src/org/compiere/model/MAttributeInstance.java

```diff
@@ -213,11 +224,10 @@ else if (displayType == DisplayType.Time)
 	 * @param valueInt
 	 * @param value
 	 */
-	public void setValueInt(int valueInt, String value)
+	public void setValueInt(int valueInt)
 	{
 		super.setValueNumber(BigDecimal.valueOf(valueInt));
-		if (value == null)
-			value = String.valueOf(valueInt);
+		String	value = String.valueOf(valueInt);

```

Comment:
just curious, what's the use case here to convert valueInt to string and pass it to the string value field ?

Reply:
@hengsin we are converting Int to String as Value column is of String type.

Reply:
yeah, I can read the code. I just don't get it why do we need to do that here ? what use case does it serve here ?

Reply:
nvm, that seems what other set method is doing as well.